app/data_providers/user_recommendation_preferences_provider.py

```diff
@@ -98,8 +104,117 @@ async def _model_from_feature_store_record(
     def _feature_store_record_from_model(cls, model: UserRecommendationPreferencesModel) -> List[Dict[str, Any]]:
         return [
             {
-                'FeatureName': 'user_id',
-                'ValueAsString': model.user_id
+                'FeatureName': cls._FEATURE_ID_NAME,
+                'ValueAsString': getattr(model, cls._FEATURE_ID_NAME)
+            },
+            {
+                'FeatureName': 'updated_at',
+                'ValueAsString': model.updated_at.strftime('%Y-%m-%dT%H:%M:%SZ')
+            },
+            {
+                'FeatureName': 'preferred_topics',
+                'ValueAsString': json.dumps([cls._topic_feature_from_topic(t) for t in model.preferred_topics])
+            },
+        ]
+
+    @classmethod
+    def _topic_feature_from_topic(cls, topic: TopicModel) -> Dict[str, str]:
+        return {'id': topic.id}
+
+
+class UserRecommendationPreferencesProviderV2:
```

Comment:
This class is duplicated from `UserRecommendationPreferencesProvider`. Originally I had created a derived class, but after deciding to rename `user_id` to `hashed_user_id`, the number of changes between the two classes grew. The v1 feature group will be decommissioned, so this duplicate code won't hand around for long.